rpcserver.go

```diff
@@ -3152,14 +3154,36 @@ func (r *rpcServer) DescribeGraph(ctx context.Context,
 		return nil, err
 	}
 
+	// Get list of private channels
+	privateChannelIDs := map[uint64]bool{}
+	if !private {
+		privateChannelIDs, err = db.FetchPrivateOpenChannelIDs()
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	// Next, for each active channel we know of within the graph, create a
 	// similar response which details both the edge information as well as
 	// the routing policies of th nodes connecting the two edges.
 	err = graph.ForEachChannel(func(edgeInfo *channeldb.ChannelEdgeInfo,
 		c1, c2 *channeldb.ChannelEdgePolicy) error {
 
-		edge := marshalDbEdge(edgeInfo, c1, c2)
-		resp.Edges = append(resp.Edges, edge)
+		// Do not include private channels unless specifically
+		// requested.
+		// A private channel does not need an authentication proof,
+		// while an advertized channel must have one. However in some
+		// situations, and specificatlly in integration tests (e.g. in
+		// open channel reorg test), public channels do not yet have
+		// an auth proof, hence relying on authProof != nil to determine
+		// if a channel is public would fail. Instead, to determine if the
```

Comment:
Feels like this should be handled by the test instead?

If we rely on `authProof` here, the integration test should be aware that the channel needs 6 confirmations before it is public.

Reply:
I guess it boils down to taste, what you feel is useful information given by the RPC call. 

To me, it sounds useful to let `describegraph` be the API for getting your nodes' "view of the network", and then checking the `authProof` would give a nice overview of which channels your node is considering "announced" (come to think of it, I don't think there exist any way currently to get this information). As mentioned, `ListChannels` can be used to get the "is private now, and staying private" information about your own channels. 

However, you have well thought of arguments, and I'm certainly not saying my opinion is the right one in this case. We can let others way in on the discussion if we don't reach an agreement 😄

Reply:
I think the right approach is to add a separate 'announced' flag which if set would return only public channels that were announced. I feel its wrong to use 'private' flag to something that is not exactly 'private'; a channel shouldn't change from being private to being public, rather a public channel state changes from not having a confirmed auth proof to having one.
I can easily add an 'announced' flag too, if you think this makes sense.

Reply:
Or maybe add an 'announced' bool property to the edges in the describegraph response, so you got the info there and can further filter if needed.

Reply:
What do you think about changing the flag's name to `included_not_announced` or similar?

Reply:
Reason being there's a similar concern (not as big as for private channels though) to people using `describegraph` and by accident leak channels that are not yet announced, as they shouldn't be used before they are properly announced.

Reply:
Yes, this sounds right to me (--include_unannounced)

In terms of integration tests:
- I'll add the flag to Alice's describeGraph query in the reorg test, and add a comment there something along: "Request all edges as channel is opened but authentication proof has not been confirmed yet"
- In the private channels test I'll leave the test as in this pull request, just change the flag name.
- I'll add a new integration test that tests the new flag.